### PR TITLE
fix: set default catalog

### PR DESCRIPTION
**Pull request by betodealmeida**
### SUMMARY

A few more small fixes for the catalog feature, ensuring that permissions work when multi-catalog is disabled and the API passes `catalog=None` to the backend.

This PR also fixes a **ton** of integration tests that rely on side-effects from other tests.

### TESTING INSTRUCTIONS

Added and updated tests.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/models/slice.py [hunk 1]
```diff
@@ -370,6 +370,7 @@ def set_related_perm(_mapper: Mapper, _connection: Connection, target: Slice) ->
         if ds:
             target.perm = ds.perm
             target.schema_perm = ds.schema_perm
+            target.catalog_perm = ds.catalog_perm
```
Comment (Vitor-Avila) [inline superset/models/slice.py:373]: TIL about this function!

**Review by Vitor-Avila — APPROVED** [review]
Thank you so much for these fixes!

File: superset/models/slice.py (hunk 1 above)
Comment (betodealmeida) [reply superset/models/slice.py:373]: Same!

**Comment by betodealmeida** [comment]
Superseded by https://github.com/apache/superset/pull/29840.

**Pull request closed without merging**